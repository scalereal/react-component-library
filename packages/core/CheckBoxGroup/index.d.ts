import { FC } from 'react';
import { Props } from './types';
declare const CheckBoxGroup: FC<Props>;
export default CheckBoxGroup;
