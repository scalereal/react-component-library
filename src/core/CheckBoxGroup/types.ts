import { ReactNode } from "react";

export type CheckBoxSizes = 'XS' | 'S' | 'M' | 'L' | 'XL' ;
export type LabelPositions = 'left'| 'right'| 'top'| 'bottom';
export type displayStyle = 'inline' | 'block';

export interface Props{
    display?: displayStyle;
    children?:ReactNode;
     /** Options */
     options: {
        value: any;
        label: string;
        showSelectAll?: boolean;
        disabled?: boolean;
        errorText?: string;
        helperText?: string;
        columns?: number;
    }[];
}