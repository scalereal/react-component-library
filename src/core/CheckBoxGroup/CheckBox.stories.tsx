import { Meta, Story } from '@storybook/react/types-6-0';
import React from 'react';
import CheckBoxGroup from '.';
import { Props } from './types';


export default {
    title: 'Components/Core/CheckBoxGroup',
    component: CheckBoxGroup,
    argTypes : {
        // size : {
        //     control: {
        //         type: 'select',
        //         options: Object.values(SizeMap),
        //     },
        // },
        // label:{
        //     control:{type:'text'}
        // },
        // disabled:{
        //     control:{type:'boolean'}
        // },
        // hasError:{
        //     control: { type: 'boolean'}
        // },
        // errorText:{
        //     control:{type:'text'}
        // },
        // bgColor : {
        //     control:'color'
        // },
        // textColor: {
        //     control:'color'
        // },
        // hoverBgColor: {
        //     control:'color'
        // },
        // hoverTextColor: {
        //     control:'color'
        // },
        // hoverShadowColor: {
        //     control:'color'
        // },
        // hoverTextShadowColor: {
        //     control:'color'
        // },
        // hoverEffect:{ 
        //     control: { type: 'boolean'}  
        // },
        
    }
} as Meta;

const Template: Story<Props> = ({...args }) => {
    // retrieves the appropriate icon passes it as a component prop
    return <CheckBoxGroup {...args} />;
};

export const Default = Template.bind({});
Default.args={
    display:'block',
    options: [
        {value: 1,label: "1"},
        {value: 2,label: "2"}
    ]
}


