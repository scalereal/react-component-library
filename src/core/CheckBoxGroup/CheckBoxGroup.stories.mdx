import CheckBoxGroup, { lposMap, SizeMap, DisplayMap } from '.';
import { Preview, Story, Meta, Props, ArgsTable, Canvas } from '@storybook/addon-docs/blocks';
import { withKnobs, color, boolean, select } from '@storybook/addon-knobs';

<Meta
    title="Components/Core/CheckBoxGroup"
    component={CheckBoxGroup}
    argTypes={{
        size: {
            control: {
                type: 'select',
                options: Object.values(SizeMap)
            }
        },
        fontSize: {
            control: {
                type: 'select',
                options: Object.values(SizeMap)
            }
        },
        label: {
            control: { type: 'text' }
        },
        labelPosition: {
            control: {
                type: 'select',
                options: Object.values(lposMap)
            }
        },
        display: {
            control: {
                type: 'select',
                options: Object.values(DisplayMap)
            }
        },
        showSelectAll: {
            control: { type: 'boolean' }
        },
        hoverEffect: {
            control: { type: 'boolean' }
        },
        labelHoverEffect: {
            control: { type: 'boolean' }
        },
        disabled: {
            control: { type: 'boolean' }
        },
        hasError: {
            control: { type: 'boolean' }
        },
        errorText: {
            control: { type: 'text' }
        },
        bgColor: {
            control: 'color'
        },
        labelColor: {
            control: 'color'
        },
        hoverBgColor: {
            control: 'color'
        },
        labelHoverColor: {
            control: 'color'
        },
        hoverShadowColor: {
            control: 'color'
        },
        ref: {
            table: {
                disable: true
            }
        },
        theme: {
            table: {
                disable: true
            }
        }
    }}
/>

# CheckBoxGroup

Displays a CheckBoxGroup

export const Template = args => (
    <div>
        <CheckBoxGroup {...args} />
    </div>
);

<Preview>
    <Story
        name="CheckBoxGroup"
        args={{
            display: 'block',
            label: 'CheckBoxGroup Label',
            size: 'S',
            checkBoxOptions: [
                { id: '1', value: 1, label: '1' },
                { id: '2', value: 2, label: '2' }
            ]
        }}
    >
        {Template.bind({})}
    </Story>
</Preview>

### Select All

Displays Select All for CheckBoxGroup.

<Preview withToolbar>
    <div>
        <CheckBoxGroup
            display="inline"
            label="CheckBoxGroup Label"
            size="S"
            checkBoxOptions={[
                { id: '1', value: 1, label: '1' },
                { id: '2', value: 2, label: '2' }
            ]}
            showSelectAll={true}
        />
    </div>
</Preview>

### Display Inline

Displays inline Checkboxes.

<Preview withToolbar>
    <div>
        <CheckBoxGroup
            display="inline"
            label="CheckBoxGroup Label"
            size="S"
            checkBoxOptions={[
                { id: '1', value: 1, label: '1' },
                { id: '2', value: 2, label: '2' }
            ]}
        />
    </div>
</Preview>

### Error

Displays error text if not valid or hasError={true}.

<Preview withToolbar>
    <div>
        <CheckBoxGroup
            display="inline"
            label="CheckBoxGroup Label"
            size="S"
            hasError={true}
            errorText="Please select any"
            checkBoxOptions={[
                { id: '1', value: 1, label: '1' },
                { id: '2', value: 2, label: '2' }
            ]}
            showSelectAll={true}
        />
    </div>
</Preview>

### Color Variations

You can either change the theme or pass the colors as props.

<Preview withToolbar>
    <div>
        <CheckBoxGroup
            display="block"
            label="CheckBoxGroup Label"
            size="S"
            checkBoxOptions={[
                { id: '1', value: 1, label: '1', checked: true },
                { id: '2', value: 2, label: '2' }
            ]}
            bgColor="#37d5d3"
            labelColor="blue"
        />
    </div>
</Preview>

### Sizes

There are five sizs available `XS` | `S` | `M` | `L` | `XL`.

<Preview withToolbar>
    <div style={{ display: 'flex', flexWrap: 'wrap', justifyContent: 'space-around' }}>
        <CheckBoxGroup
            display="inline"
            label="CheckBoxGroup Label"
            size="XS"
            checkBoxOptions={[
                { id: '1', value: 1, label: '1' },
                { id: '2', value: 2, label: '2' }
            ]}
            bgColor="#37d5d3"
            labelColor="blue"
        />
        <CheckBoxGroup
            display="inline"
            label="CheckBoxGroup Label"
            size="S"
            checkBoxOptions={[
                { id: '1', value: 1, label: '1' },
                { id: '2', value: 2, label: '2' }
            ]}
            bgColor="#37d5d3"
            labelColor="blue"
        />
        <CheckBoxGroup
            display="inline"
            label="CheckBoxGroup Label"
            size="M"
            checkBoxOptions={[
                { id: '1', value: 1, label: '1' },
                { id: '2', value: 2, label: '2' }
            ]}
            bgColor="#37d5d3"
            labelColor="blue"
        />
        <CheckBoxGroup
            display="inline"
            label="CheckBoxGroup Label"
            size="L"
            checkBoxOptions={[
                { id: '1', value: 1, label: '1' },
                { id: '2', value: 2, label: '2' }
            ]}
            bgColor="#37d5d3"
            labelColor="blue"
        />
        <CheckBoxGroup
            display="inline"
            label="CheckBoxGroup Label"
            size="XL"
            checkBoxOptions={[
                { id: '1', value: 1, label: '1' },
                { id: '2', value: 2, label: '2' }
            ]}
            bgColor="#37d5d3"
            labelColor="blue"
        />
    </div>
</Preview>

### Text Sizes

There are five Text sizs available `XS` | `S` | `M` | `L` | `XL`.

<Preview withToolbar>
    <div style={{ display: 'flex', flexWrap: 'wrap', justifyContent: 'space-around' }}>
        <CheckBoxGroup
            display="inline"
            label="CheckBoxGroup Label"
            fontSize="XS"
            checkBoxOptions={[
                { id: '1', value: 1, label: '1' },
                { id: '2', value: 2, label: '2' }
            ]}
            bgColor="#37d5d3"
            labelColor="blue"
        />
        <CheckBoxGroup
            display="inline"
            label="CheckBoxGroup Label"
            fontSize="S"
            checkBoxOptions={[
                { id: '1', value: 1, label: '1' },
                { id: '2', value: 2, label: '2' }
            ]}
            bgColor="#37d5d3"
            labelColor="blue"
        />
        <CheckBoxGroup
            display="inline"
            label="CheckBoxGroup Label"
            fontSize="M"
            checkBoxOptions={[
                { id: '1', value: 1, label: '1' },
                { id: '2', value: 2, label: '2' }
            ]}
            bgColor="#37d5d3"
            labelColor="blue"
        />
        <CheckBoxGroup
            display="inline"
            label="CheckBoxGroup Label"
            fontSize="L"
            checkBoxOptions={[
                { id: '1', value: 1, label: '1' },
                { id: '2', value: 2, label: '2' }
            ]}
            bgColor="#37d5d3"
            labelColor="blue"
        />
        <CheckBoxGroup
            display="inline"
            label="CheckBoxGroup Label"
            fontSize="XL"
            checkBoxOptions={[
                { id: '1', value: 1, label: '1' },
                { id: '2', value: 2, label: '2' }
            ]}
            bgColor="#37d5d3"
            labelColor="blue"
        />
    </div>
</Preview>

### Props

<Props story="CheckBoxGroup" />
