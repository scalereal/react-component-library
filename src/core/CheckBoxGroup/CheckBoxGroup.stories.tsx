import { Meta, Story } from '@storybook/react/types-6-0';
import React from 'react';
import CheckBoxGroup from '.';
import { Props } from './types';

const SizeMap: Props['size'][] = ['XS', 'S', 'M', 'L', 'XL'];
const DisplayMap: Props['display'][] = ['inline', 'block'];
const lposMap: Props['labelPosition'][] = ['top','bottom','left','right']

export default {
    title: 'Components/Core/CheckBoxGroup',
    component: CheckBoxGroup,
    argTypes : {
        size : {
            control: {
                type: 'select',
                options: Object.values(SizeMap),
            },
        },
        fontSize:{
            control: {
                type: 'select',
                options: Object.values(SizeMap),
            },
        },
        label:{
            control:{type:'text'}
        },
        labelPosition:{
            control: {
                type: 'select',
                options: Object.values(lposMap),
            },
        },
        display:{
            control: {
                type: 'select',
                options: Object.values(DisplayMap),
            },
        },
        showSelectAll:{ 
            control: { type: 'boolean'}  
        },
        hoverEffect:{ 
            control: { type: 'boolean'}  
        },
        labelHoverEffect:{ 
            control: { type: 'boolean'}  
        },
        disabled:{
            control:{type:'boolean'}
        },
        hasError:{
            control: { type: 'boolean'}
        },
        errorText:{
            control:{type:'text'}
        },
        bgColor : {
            control:'color'
        },
        labelColor: {
            control:'color'
        },
        hoverBgColor: {
            control:'color'
        },
        labelHoverColor: {
            control:'color'
        },
        hoverShadowColor: {
            control:'color'
        },
        ref:{
            table:{
              disable:true
            }
        },
        theme:{
            table:{
              disable:true
            }
        },
    }
} as Meta;

const Template: Story<Props> = ({...args }) => {
    // retrieves the appropriate icon passes it as a component prop
    return <CheckBoxGroup {...args} />;
};

export const Default = Template.bind({});
Default.args={
    display:'block',
    label:"CheckBoxGroup Label",
    size:'S',
    checkBoxOptions: [
        {id:"1",value: 1,label: "1"},
        {id:"2",value: 2,label: "2"}
    ],
    showSelectAll:true
}


