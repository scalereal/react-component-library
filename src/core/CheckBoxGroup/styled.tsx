import styled from 'styled-components';
import { Props } from './types';

// export const ErrorLabel = styled.label<Props>`
//     display:block;
//     color: ${defaultTheme.statusColors.danger}
// `;

// export const StyledLabel = styled.label<Props>`
//     display: inline-flex;
//     cursor: pointer;
//     padding: 0.8rem 0px;
//     align-items: center;
//     justify-content: flex-end;
//     user-select: none;
//     ${pr => pr.disabled === true?'pointer-events: none;opacity: 0.4;':''}
//     span{
//         margin: 0px;
//         color: inherit;
//         font-size: 1.4rem;
//         font-weight: 400;
//         letter-spacing: 0rem;
//         line-height: 2.2rem;
//         text-align: initial;
//         padding: 0px 1.6rem;    
//         }

//     input[type="checkbox"] {
//         border: 1px solid ${pr => pr.hasError?'#CC0000': defaultTheme.borderColor };
//         -webkit-appearance: none;
//         -webkit-transition: box-shadow 200ms;
        
//         ${pr => pr.hoverEffect? `&:hover {
//             input[type="checkbox"]:checked{
//                 background-color: ${pr.hoverBgColor || defaultTheme.primaryActiveColor}
//                 };
//             border-color:${defaultTheme.primaryColor};
//             box-shadow: 0px 4px 19px ${pr.hoverShadowColor || defaultTheme.shadowColor};
//             }
//             `:
//             ''
//             } 
//     }

//     /* border radius for checkbox */
//     input[type="checkbox"] {
//         -webkit-border-radius:5px;
//         border-radius:5px;
//     }

//     /* focus state */
//     input[type="checkbox"]:focus {
//         outline:none;
//     }

//     /* input checked border color */
//     input[type="checkbox"]:checked {
//         background-color: ${pr => pr.hasError?'#CC0000': defaultTheme.primaryColor };
//     }

//     input[type="checkbox"]:checked:before {
//         content: '';
//         display: block;
//         border: solid #fff;
//         -webkit-transform: rotate(45deg);
//         transform: rotate(45deg);
//         margin:auto;
//     }
//     ${pr=>  mixin(pr.size)}
        
// `;

export const StyledLi = styled.li`
    width:auto;
`;

export const StyledUl = styled.ul<Props>`
    display:flex;
    flex-direction:row;
    margin:0;
    padding:0;
    list-style:none;
    ${pr => 
        pr.display==='block'? `flex-direction:column;${StyledLi}{
        width:100%;
        }`:''
    }
`;

