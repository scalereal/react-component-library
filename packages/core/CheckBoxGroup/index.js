var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import React, { memo, useState } from 'react';
import CheckBox from '../CheckBox';
import { CheckBoxWrapper, ErrorLabel } from '../CheckBox/styled';
import Text from '../Text/Text';
import { StyledLabel, StyledLi, StyledUl } from './styled';
var CheckBoxGroup = memo(React.forwardRef(function (props, ref) {
    var _a = props.display, display = _a === void 0 ? "block" : _a, checkBoxOptions = props.checkBoxOptions, errorText = props.errorText, size = props.size, fontSize = props.fontSize, labelColor = props.labelColor, labelHoverEffect = props.labelHoverEffect, labelHoverColor = props.labelHoverColor, label = props.label, _b = props.showSelectAll, showSelectAll = _b === void 0 ? false : _b, hoverEffect = props.hoverEffect, disabled = props.disabled, bgColor = props.bgColor, hoverBgColor = props.hoverBgColor, hoverShadowColor = props.hoverShadowColor, labelPosition = props.labelPosition;
    var _c = useState(false), selectAll = _c[0], setSelectAll = _c[1];
    return (React.createElement(CheckBoxWrapper, __assign({ ref: ref }, props),
        React.createElement(ErrorLabel, { id: "error", htmlFor: "error" }, errorText),
        React.createElement(StyledLabel, __assign({ size: size }, props),
            showSelectAll ? React.createElement("input", { type: "checkbox", checked: selectAll, onChange: function () { return setSelectAll(!selectAll); } }) : '',
            React.createElement(Text, { textSize: fontSize, textColor: labelColor, hoverEffect: labelHoverEffect, hoverColor: labelHoverColor }, label)),
        React.createElement(StyledUl, __assign({ display: display }, props), checkBoxOptions && checkBoxOptions.map(function (obj) {
            return (React.createElement(StyledLi, { key: obj.id },
                React.createElement(CheckBox, { key: obj.id, label: obj.label, value: obj.value, checked: obj.checked ? obj.checked : selectAll, size: size, fontSize: fontSize, labelColor: labelColor, hoverEffect: hoverEffect, labelHoverEffect: labelHoverEffect, labelHoverColor: labelHoverColor, disabled: obj.disabled ? obj.disabled : disabled, bgColor: bgColor, hoverBgColor: hoverBgColor, hoverShadowColor: hoverShadowColor, labelPosition: labelPosition })));
        }))));
}));
CheckBoxGroup.displayName = "CheckBoxGroup";
CheckBoxGroup.defaultProps = {
// hoverEffect: false
};
export default CheckBoxGroup;
