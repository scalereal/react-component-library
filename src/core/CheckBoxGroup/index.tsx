import React, { FC, memo, useState } from 'react';
import CheckBox from '../CheckBox';
import { CheckBoxWrapper, ErrorLabel } from '../CheckBox/styled';
import Text from '../Text/Text';
import { StyledLabel, StyledLi, StyledUl } from './styled';
import { Props } from './types';

export const SizeMap: Props['size'][] = ['XS', 'S', 'M', 'L', 'XL'];
export const DisplayMap: Props['display'][] = ['inline', 'block'];
export const lposMap: Props['labelPosition'][] = ['top','bottom','left','right']

const CheckBoxGroup:FC<Props> = memo(React.forwardRef( (props, ref) => {
    const {display="block",checkBoxOptions,errorText,size,
    fontSize,labelColor,labelHoverEffect,labelHoverColor,
    label,showSelectAll=false,hoverEffect,disabled,
    bgColor,hoverBgColor,hoverShadowColor,labelPosition} = props;

    const [selectAll,setSelectAll] = useState<boolean>(false);

    return (
        <CheckBoxWrapper ref={ref} {...props}>
            <ErrorLabel id="error" htmlFor="error">{errorText}</ErrorLabel>
            <StyledLabel size={size} {...props} >
                {showSelectAll?<input type="checkbox" checked={selectAll} onChange={()=>setSelectAll(!selectAll)}/>:''}
                <Text textSize={fontSize} textColor={labelColor} hoverEffect={labelHoverEffect} hoverColor={labelHoverColor} >{label}</Text>
            </StyledLabel>
            <StyledUl display={display} {...props}>
                {checkBoxOptions && checkBoxOptions.map(obj=>{ 
                return (
                    <StyledLi key={obj.id}>
                        <CheckBox key={obj.id} label={obj.label} 
                        value={obj.value} checked={obj.checked?obj.checked:selectAll} 
                        size={size} fontSize={fontSize} labelColor={labelColor}
                        hoverEffect={hoverEffect} labelHoverEffect={labelHoverEffect} labelHoverColor={labelHoverColor}
                        disabled={obj.disabled?obj.disabled:disabled} bgColor={bgColor}
                        hoverBgColor={hoverBgColor} hoverShadowColor={hoverShadowColor}
                        labelPosition={labelPosition}
                        />
                    </StyledLi>)
                })}
            </StyledUl>
        </CheckBoxWrapper>
        )
}));

CheckBoxGroup.displayName="CheckBoxGroup";
CheckBoxGroup.defaultProps = {
    hoverEffect: true,
    size:'S'
};

export default CheckBoxGroup;
