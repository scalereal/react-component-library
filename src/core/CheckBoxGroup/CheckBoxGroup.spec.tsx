import { render } from '@testing-library/react';
import * as React from 'react';
import CheckBoxGroup from '.';

describe('CheckBoxGroup component', () => {
    it('should render with default theme', () => {
        const { container } = render(<CheckBoxGroup/>);
        expect(container).toMatchSnapshot();
    });

    it('should render with all the props given', () => {
        const { container } = render(
            <CheckBoxGroup
                label="checkBoxGroup Label"
                size="M"
                fontSize='M'
                labelColor='white'
                bgColor= 'blue'
                showSelectAll={true}
                hoverEffect={true}
                labelHoverEffect={true}
                hoverBgColor='#126AFA'
                labelHoverColor= '#12fadb'
                hoverShadowColor="rgba(96, 120, 144, 0.35)"
                checkBoxOptions={[
                    {id:"1",value: 1,label: "1"},
                    {id:"2",value: 2,label: "2"}
                ]}
            />
            );
        expect(container).toMatchSnapshot();
    });

    it('should render with not-allowed cursor when it is disabled', () => {
        const { container } = render(
            <CheckBoxGroup disabled={true} />
        );
        expect(container).toMatchSnapshot();
    });

});