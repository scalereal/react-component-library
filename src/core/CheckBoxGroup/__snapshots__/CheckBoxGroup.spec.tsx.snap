// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBoxGroup component should render with all the props given 1`] = `
<div>
  <div
    class="sc-hKgILt cTJgxF"
    font-size="M"
    label="checkBoxGroup Label"
  >
    <label
      class="sc-gsTCUz jbqcjW"
      for="error"
      id="error"
    />
    <label
      class="sc-gKsewC fDMCFB"
      font-size="M"
      label="checkBoxGroup Label"
    >
      <input
        type="checkbox"
      />
      <span
        class="sc-bdfBwQ bLBrZi"
      >
        checkBoxGroup Label
      </span>
    </label>
    <ul
      class="sc-jSgupP dPSTHG"
      display="block"
      font-size="M"
      label="checkBoxGroup Label"
    >
      <li
        class="sc-eCssSg jZAUgl"
      >
        <div
          class="sc-hKgILt cTJgxF"
          font-size="M"
          label="1"
          value="1"
        >
          <label
            class="sc-gsTCUz jbqcjW"
            for="error"
            id="error"
          />
          <label
            class="sc-dlfnbm UwbaL"
            font-size="M"
            label="1"
            value="1"
          >
            <input
              type="checkbox"
              value="1"
            />
            <span
              class="sc-bdfBwQ bLBrZi"
            >
              1
            </span>
          </label>
        </div>
      </li>
      <li
        class="sc-eCssSg jZAUgl"
      >
        <div
          class="sc-hKgILt cTJgxF"
          font-size="M"
          label="2"
          value="2"
        >
          <label
            class="sc-gsTCUz jbqcjW"
            for="error"
            id="error"
          />
          <label
            class="sc-dlfnbm UwbaL"
            font-size="M"
            label="2"
            value="2"
          >
            <input
              type="checkbox"
              value="2"
            />
            <span
              class="sc-bdfBwQ bLBrZi"
            >
              2
            </span>
          </label>
        </div>
      </li>
    </ul>
  </div>
</div>
`;

exports[`CheckBoxGroup component should render with default theme 1`] = `
<div>
  <div
    class="sc-hKgILt cTJgxF"
  >
    <label
      class="sc-gsTCUz jbqcjW"
      for="error"
      id="error"
    />
    <label
      class="sc-gKsewC hOUZyW"
    >
      
      <span
        class="sc-bdfBwQ hcPyHd"
      />
    </label>
    <ul
      class="sc-jSgupP dPSTHG"
      display="block"
    />
  </div>
</div>
`;

exports[`CheckBoxGroup component should render with not-allowed cursor when it is disabled 1`] = `
<div>
  <div
    class="sc-hKgILt cTJgxF"
    disabled=""
  >
    <label
      class="sc-gsTCUz jbqcjW"
      for="error"
      id="error"
    />
    <label
      class="sc-gKsewC fVwTbS"
      disabled=""
    >
      
      <span
        class="sc-bdfBwQ hcPyHd"
      />
    </label>
    <ul
      class="sc-jSgupP dPSTHG"
      disabled=""
      display="block"
    />
  </div>
</div>
`;
